### PR TITLE
Parametric LLSModels

### PR DESCRIPTION
**Pull request by geoffroyleconte**
Follows the update of NLPModels: JuliaSmoothOptimizers/NLPModels.jl#353

### CODE REVIEW

File: test/test_lls_model.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@ function lls_test()
     for A in [Matrix(1.0I, 10, 3) .+ 1, sparse(1.0I, 10, 3) .+ 1],
       C in [ones(1, 3), [ones(1, 3); -I], sparse(ones(1, 3))]
 
-      b = collect(1:10)
+      b = Float64.(collect(1:10))
```
Comment (abelsiqueira) [inline test/test_lls_model.jl:6]:
```suggestion
      b = collect(1.0:10.0)
```

**Comment by abelsiqueira** [comment]
Thanks. I made a modification by hand and already committed it, because it was small.
The `docs` are failing probably because of the `Project.toml`.

**Comment by geoffroyleconte** [comment]
Thanks that's what I was wondering.

File: src/lls_model.jl [hunk 2]
```diff
@@ -311,6 +311,6 @@ function NLPModels.ghjvprod!(
   @lencheck nls.meta.nvar x g v
   @lencheck nls.meta.ncon gHv
   increment!(nls, :neval_hprod)
-  gHv .= zeros(T, nls.meta.ncon)
+  gHv .= zero(T)
```
Comment (geoffroyleconte) [inline src/lls_model.jl:314]: I change to this instead of `fill!(S(undef, nls.meta.ncon), zero(T))` I hope I am not mistaken

Comment (abelsiqueira) [reply src/lls_model.jl:314]: This is fine

**Review by abelsiqueira — APPROVED** [review]

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**